store/tikv/2pc.go

```diff
@@ -874,7 +875,7 @@ func sendTxnHeartBeat(bo *Backoffer, store *tikvStore, primary []byte, startTS,
 		}
 		cmdResp := resp.Resp.(*pb.TxnHeartBeatResponse)
 		if keyErr := cmdResp.GetError(); keyErr != nil {
-			return 0, errors.Errorf("txn %d heartbeat fail, primary key = %v, err = %s", startTS, primary, keyErr.Abort)
+			return 0, errors.Errorf("txn %d heartbeat fail, primary key = %v, err = %s", startTS, hex.EncodeToString(primary), extractKeyErr(keyErr))
```

Comment:
Maybe it's better to use `prettyWriteKey` to print?

Reply:
I think we don't care what the key represents(a concrete table key or index key) in this case. Hex format is enough and easy for debug.

Reply:
Makes sense.